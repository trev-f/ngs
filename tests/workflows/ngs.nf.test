nextflow_workflow {

    name "Test Workflow NGS"
    script "workflows/ngs.nf"
    workflow "NGS"

    test("successfully runs with default params of nf_test profile.") {

        when {
            params {}
            workflow {}
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 16
        }

    }

    test("successfully runs with uncompressed reference and annotation files.") {

        when {
            params {
                genome      = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                annotations = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
            }
            workflow {}
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 14
        }

    }

}
