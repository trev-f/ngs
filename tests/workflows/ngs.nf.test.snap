{
    "successfully runs with uncompressed reference and annotation files.": {
        "content": [
            {
                
            }
        ],
        "timestamp": "2023-04-20T18:21:54+0000"
    },
    "successfully runs with default params of nf_test profile.": {
        "content": [
            {
                
            }
        ],
        "timestamp": "2023-04-20T18:21:54+0000"
    },
    "NGS succeeds with default params.": {
        "content": [
            {
                
            }
        ],
        "timestamp": "2023-04-20T18:21:54+0000"
    }
}